File: Java/Polymorphism/shapes/Rectangle.java
package shapes;

public class Rectangle extends Shape {
    private double width;
    private double height;

    public Rectangle(double width, double height) {
        this.width = width;
        this.height = height;
    }

    @Override
    double calculatePerimeter() {
        return 2 * this.width + 2 * this.height;
    }

    @Override
    double calculateArea() {
        return this.width * this.height;
    }
}

File: JS/composition/exercise/rectangle.js
function rectangle(width, height, color) {
    function capitalize(word) {
        return word[0].toUpperCase() + word.slice(1);
    }

    const rect = {
        width: Number(width),
        height: Number(height),
        color: capitalize(color),
        calcArea() {
            let res = width * height;
            return res;
    }
    };

    return rect;
}

let rect = rectangle(4, 5, 'red');
console.log(rect.width);
console.log(rect.height);
console.log(rect.color);
console.log(rect.calcArea());
